handler.py

```diff
@@ -31,6 +46,46 @@ def handle_healthcheck_command():
     return {"command": "healthcheck", "result": healthcheck()}
 
 
+def handle_preview_command(event):
+    """
+    Function to create the response when commanded to preview an answer.
+    ---
+    This function attempts to parse the request body, performs schema validation and
+    attempts to run the evaluation function on the given parameters.
+
+    If any of these fail, a message is returned and an error field is passed if more
+    information can be provided.
+    """
+    body, parse_error = parse.parse_body(event)
+
+    if parse_error:
+        return {"error": parse_error}
+
+    request_error = v.validate_preview_request(body)
+
+    if request_error:
+        return {"error": request_error}
+
+    response = body["response"]
+    params = body.get("params", dict())
+
+    try:
+        result = preview_function(response, params)
+    # Catch the custom EvaluationException (from evaluation_function_utils) first
+    except EvaluationException as e:
+        return {"error": e.error_dict}
+
+    except Exception as e:
+        return {
+            "error": {
+                "message": "An exception was raised while executing the preview function.",
+                "detail": str(e) if str(e) != "" else repr(e),
```

Comment:
This line is repeated quite often. It would be worth to extract it into a function, like:

```python
def error_detail(error):
    return str(error) if str(error != "" else repr(error)
```